frontend/lib/screens/new_student/local_widget/student_form.dart

```diff
@@ -0,0 +1,209 @@
+                    ],
+                  ),
+                  child: Padding(
+                    padding: EdgeInsets.symmetric(
+                      horizontal: 20.0,
+                      vertical: 5.0,
+                    ),
+                    child: Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        _createnameField(student),
+                        _createsurnameField(student),
+                        _createemailField(student),
+                        _createDescriptionField(student),
+                        _createphoneField(student),
+                      ],
+                    ),
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 16.0),
+                  child: RaisedButton(
+                    color: Colors.grey[600],
+                    onPressed: () {
+                      // Validate will return true if the form is valid, or false if
+                      // the form is invalid.
+                      if (_formKey.currentState.validate()) {
+                        widget.changeStep(false);
+                        
+                      }
+                    },
+                    child: Text(
+                      'Continue',
+                      style: TextStyle(color: Colors.white),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _createnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Name',
+      ),
+      initialValue: student.name ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.name = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student name';
+        }
+        return null;
+      },
+    );
+  }
+
+Widget _createsurnameField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'Surname',
+      ),
+      initialValue: student.surname ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.surname = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student surname';
+        }
+        return null;
+      },
+    );
+  }
+
+  Widget _createemailField(Student student) {
+    return TextFormField(
+      decoration: const InputDecoration(
+        hintText: 'student@school.com',
+      ),
+      initialValue: student.email ?? '',
+      onChanged: (value) {
+        setState(() {
+          student.email = value;
+        });
+      },
+      validator: (value) {
+        if (value.isEmpty) {
+          return 'Please enter a student email';
+        }
+        return null;
+      },
```

Comment:
@BassamZ would it be possible to add an email validation with a regex? Something like this should work: 
^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$

Reply:
You can check how to do a regex validation in Flutter [here](https://stackoverflow.com/questions/56390839/flutter-regex-in-textformfield)